src/main/java/Duke.java

```diff
@@ -1,10 +1,260 @@
+    //             c.execute(tasks, ui, storage);
+    //             isExit = c.isExit();
+    //         } catch (DukeException e) {
+    //             ui.showError(e.getMessage());
+    //         } finally {
+    //             ui.showLine();
+    //         }
+    //     }
+    // }
+
+    // public static void main(String[] args) {
+    //     new Duke(FILE_PATH).run();
+    // }
+
+
+
+    //Initialize create a arraylist of tasks
+    public static ArrayList<Task> tasks = new ArrayList<Task>();
+
+
+    
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        //read the file
+        tasks = Storage.readFile(FILE_PATH);
+
+        //Print the welcome message
+        Ui.printWelcomeMessage();
+        
+
+        //Get user input
+        Scanner userScan = new Scanner(System.in);  // Create scanner object
+        String userInput = userScan.nextLine();  // Get user input
+
+        //Check the arguments provided unless it is "bye" which quits the program
+        while( !(Check.isBye(userInput)) ){
+            
+            try{
+                //If userInput is "list" print all tasks
+                if(Check.isList(userInput)){
+                    //Print out the list of tasks
+                    Ui.printTaskList(tasks);
+                    //Get user input again
+                    userInput = userScan.nextLine();  
+                }
+
+                //If userInput is "unmark" get the task number and unmark the task as done
+                else if(Check.isUnmark(userInput)){
+                    //get the task number
+                    int taskNumber = Integer.parseInt(userInput.substring(7));
+                    //if the task number is greater than the task count throw an exception
+                    if(taskNumber>tasks.size()){
+                        throw new IllegalArgumentException("The task number is greater than the number of tasks.");
+                    }
+                    //if the task number is less than 1 throw an exception
+                    if(taskNumber<1){
+                        throw new IllegalArgumentException("The task number is less than 1.");
+                    }
+                    //if there is no task at the task number throw an exception
+                    if(tasks.get(taskNumber-1)==null){
+                        throw new IllegalArgumentException("There is no task at the task number.");
+                    }
+                    //if the task is already not done throw an exception
+                    if(tasks.get(taskNumber-1).isDone==false){
+                        throw new IllegalArgumentException("The task is already not done.");
+                    }
```

Comment:
For all your error messages, you might want to consider refactoring them into another class/enum.